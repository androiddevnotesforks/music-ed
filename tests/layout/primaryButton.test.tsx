import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"
import PrimaryButton from "../../components/layout/primaryButton";

const mockProps = {
  text: "Mock Text",
  dataTestId: "Mock ID"
}

describe("PrimaryButton component", () => {
  beforeEach(() => {
    render(<PrimaryButton {...mockProps}/>)
  })
  it("Renders", () => {
    const primaryBtn = screen.getByTestId(mockProps.dataTestId)
    expect(primaryBtn).toBeInTheDocument()
  })
  it("Matches snapshot", () => {
    const primaryBtn = screen.getByTestId(mockProps.dataTestId)
    expect(primaryBtn).toMatchSnapshot()
  })
  it("Text prop is stated", () => {
    const primaryBtn = screen.getByTestId(mockProps.dataTestId)
    expect(primaryBtn.textContent).toMatch(mockProps.text)
  })
})