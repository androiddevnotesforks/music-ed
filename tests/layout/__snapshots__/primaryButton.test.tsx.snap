// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton component Matches snapshot 1`] = `
<button
  class="undefined rounded border p-4 m-2 font-extrabold text-center hover:bg-blue-50 active:bg-blue-100 hover:cursor-pointer"
  data-testid="Mock ID"
>
  Mock Text
</button>
`;
